includes/Modules/AdSense/AMP_Tag.php

```diff
@@ -66,6 +66,33 @@ public function register() {
 		$this->do_init_tag_action();
 	}
 
+	/**
+	 * Gets the attributes for amp-story-auto-ads and amp-auto-ads tags.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $filter The filter we want to apply.
+	 * @return array Filtered $options.
+	 */
+	private function get_auto_ads_attributes( $filter ) {
+		$options = array(
+			'ad-client' => $this->tag_id,
+		);
+
+		if ( ! empty( $this->story_ad_slot_id ) ) {
+			$options['ad-slot'] = $this->story_ad_slot_id;
+		}
+
+		$filtered_options = apply_filters( $filter, $options, $this->tag_id, $this->story_ad_slot_id );
```

Comment:
:man_facepalming: ah... I forgot that methods use different filter names... Ok, let's do it this way then:

```php
private function get_auto_ads_attributes( $type = '' ) {
    ...
    if ( 'web-story' === $type && ! empty( $this->story_ad_slot_id ) ) {
        $options['ad-slot'] = $this->story_ad_slot_id;
    }

    $filtered_options = 'web-story' === $type
        ? apply_filters( 'googlesitekit_amp_story_auto_ads_attributes', $options, $this->tag_id, $this->story_ad_slot_id )
        : apply_filters( 'googlesitekit_amp_auto_ads_attributes', $options, $this->tag_id, $this->story_ad_slot_id );
    ...
}

protected function render() {
    ...
    foreach ( $this->get_auto_ads_attributes() as $amp_auto_ads_opt_key => $amp_auto_ads_opt_value ) {
        ...
    }
    ...
}

private function render_story_auto_ads() {
    ...
    foreach ( $this->get_auto_ads_attributes( 'web-story' ) as $amp_auto_ads_opt_key => $amp_auto_ads_opt_value ) {
        ...
    }
    ...
}
```

Reply:
Thanks @eugene-manuilov Since we are calling `apply_filters` with the same set of args, I slightly tweaked your approach to just have the `apply_filters` call once. Let me know what you think.